st3/sublime_lib/resource_path.py

```diff
@@ -82,9 +94,27 @@ def __str__(self):
     def __eq__(self, other):
         return isinstance(other, ResourcePath) and self._parts == other.parts
 
+    def _sort_key(self):
+        return (
+            ROOT_ORDER.get(self.root, 0),
+            self.root,
+            PACKAGE_ORDER.get(self.package, 0),
+            self._parts[1:]
```

Comment:
Something we haven't considered yet: Resource paths are sorted files-first, i.e. all files come before any subdirectories.

```py
>>> sublime.find_resources("test.test")
['Packages/User/Test/test.test', 'Packages/User/Test/A/test.test']
```

And it's case-insensitive:
```py
>>> sublime.find_resources("test.test")
['Packages/User/Test/test.test', 'Packages/User/Test/A/test.test', 'Packages/User/Test/a/test.test', 'Packages/User/Test/B/test.test']
```

Also, this should start at `[2:]`.

Reply:
It looks like it's even more complicated than that. Physical files in `Packages` seem to be sorted after resources in `Installed Packages`, and I have a sneaking suspicion that resources in zipfiles are loaded in the zipfile directory order. I will investigate this probably tomorrow.